File: server/python/app/infrastructure/mappers/payment_details/operation_payment_references_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.operation_payment_references import OperationPaymentReferences
from onlinepayments.sdk.domain.operation_payment_references import OperationPaymentReferences as OperationPaymentReferencesSdk

class OperationPaymentReferencesMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[OperationPaymentReferencesSdk]) -> OperationPaymentReferences:

        dto = OperationPaymentReferences()
        dto.merchant_reference = response.merchant_reference

        return dto

File: server/ruby/app/infrastructure/mappers/payment_details/operation_payment_references_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class OperationPaymentReferencesMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::OperationPaymentReferences.new
          dto.merchant_reference = response.merchant_reference

          dto
        end
      end
    end
  end
end